### PR TITLE
Fix links not being keyboard accessible

### PR DESCRIPTION
**Pull request by rezbyte**
Extracted from #6334.

### CODE REVIEW

File: src/misc/news/items/ReferralLinkViewer.ts [hunk 1]
```diff
@@ -36,11 +37,7 @@ export class ReferralLinkViewer implements Component<ReferralLinkAttrs> {
 			label: "referralLink_label",
 			value: referralLink,
 			injectionsRight: () => this.renderButtons(referralLink),
-			helpLabel: () =>
-				ifAllowedTutaLinks(locator.logins, InfoLink.ReferralLink, (link) => [
-					m("span", lang.get("moreInfo_msg") + " "),
-					m("span.text-break", [m(`a[href=${link}][target=_blank]`, link)]),
-				]),
+			helpLabel: () => ifAllowedTutaLinks(locator.logins, InfoLink.ReferralLink, (link) => [m(MoreInfoLink, { link: link as InfoLink })]),
```
Comment (charlag) [inline src/misc/news/items/ReferralLinkViewer.ts:40]: we should change  type for `ifAllowedTutaLinks`

should be:

```ts
export function ifAllowedTutaLinks(logins: LoginController, linkId: InfoLink, render: (linkId: InfoLink) => Children): Children | null
```

then you don't need a cast

File: src/settings/DesktopSettingsViewer.ts [hunk 2]
```diff
@@ -88,8 +89,7 @@ export class DesktopSettingsViewer implements UpdatableSettingsViewer {
 			helpLabel: () => {
 				return ifAllowedTutaLinks(locator.logins, InfoLink.RunInBackground, (link) => [
 					m("span", lang.get("runInBackground_msg") + " "),
-					m("span", lang.get("moreInfo_msg") + " "),
-					m("span.text-break", [m(`a[href=${link}][target=_blank]`, link)]),
+					m(MoreInfoLink, { link: link as InfoLink }),
```
Comment (charlag) [inline src/settings/DesktopSettingsViewer.ts:92]: you wouldn't need cast here either

File: src/misc/passwords/PasswordGeneratorDialog.ts [hunk 3]
```diff
@@ -73,15 +74,11 @@ class PasswordGeneratorDialog implements Component<PasswordGeneratorDialogAttrs>
 			m(".small.mt-xs", [
 				lang.get("passphraseGeneratorHelp_msg"),
 				" ",
-				m(
-					"a",
-					{
-						href: "https://tuta.com/faq#passphrase-generator",
-						target: "_blank",
-						rel: "nooopener noreferer",
```
Comment (charlag) [inline src/misc/passwords/PasswordGeneratorDialog.ts:81]: we've lost some `nooopener noreferer` in places, do we need them?

File: src/settings/login/LoginSettingsViewer.ts [hunk 4]
```diff
@@ -98,10 +99,7 @@ export class LoginSettingsViewer implements UpdatableSettingsViewer {
 		const recoveryCodeFieldAttrs: TextFieldAttrs = {
 			label: "recoveryCode_label",
 			helpLabel: () => {
-				return ifAllowedTutaLinks(locator.logins, InfoLink.RecoverCode, (link) => [
-					m("span", lang.get("moreInfo_msg") + " "),
-					m("span.text-break", [m(`a[href=${link}][target=_blank]`, link)]),
-				])
+				return ifAllowedTutaLinks(locator.logins, InfoLink.RecoverCode, (link) => [m(MoreInfoLink, { link: link as InfoLink })])
```
Comment (charlag) [inline src/settings/login/LoginSettingsViewer.ts:102]: wouldn't need a cast in these two places

File: src/settings/login/secondfactor/SecondFactorsEditForm.ts [hunk 5]
```diff
@@ -50,12 +50,7 @@ export class SecondFactorsEditForm {
 			m(".h4.mt-l", lang.get("secondFactorAuthentication_label")),
 			m(Table, secondFactorTableAttrs),
 			this.domainConfigProvider.getCurrentDomainConfig().firstPartyDomain
-				? [
-						m("span.small", lang.get("moreInfo_msg") + " "),
-						ifAllowedTutaLinks(locator.logins, InfoLink.SecondFactor, (link) =>
-							m("span.small.text-break", [m(`a[href=${link}][target=_blank]`, link)]),
-						),
-				  ]
+				? [ifAllowedTutaLinks(locator.logins, InfoLink.SecondFactor, (link) => m(MoreInfoLink, { link: link as InfoLink, isSmall: true }))]
```
Comment (charlag) [inline src/settings/login/secondfactor/SecondFactorsEditForm.ts:53]: or here

File: src/misc/passwords/PasswordGeneratorDialog.ts (hunk 3 above)
Comment (rezbyte) [reply src/misc/passwords/PasswordGeneratorDialog.ts:81]: We might actually. I mistakenly thought `target: "_blank"` automatically includes both `noreferrer` & `noopener`. `target: "_blank"`  [automatically includes `noopener`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noopener) on [all browsers except Android Webview](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#browser_compatibility). `noreferrer`  [automatically includes `noopener`](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noreferrer) on all browsers. So I will add just `noreferrer` back where needed.

File: src/misc/news/items/ReferralLinkViewer.ts (hunk 1 above)
Comment (rezbyte) [reply src/misc/news/items/ReferralLinkViewer.ts:40]: Understood, will do!

**Review by charlag — APPROVED** [review]

**Pull request merged**